app/components/viral/modal_component.html.erb

```diff
@@ -0,0 +1,99 @@
+<div data-controller="viral--modal-component">
+  <!-- Modal toggle -->
+  <button
+    data-viral--modal-component-target="button"
+    data-action="click->viral--modal-component#open"
+    class="
+      block
+      text-white
+      bg-blue-700
+      hidden
+      hover:bg-blue-800
+      focus:ring-4
+      focus:outline-none
+      focus:ring-blue-300
+      font-medium
+      rounded-lg
+      text-sm
+      px-5
+      py-2.5
+      text-center
+      dark:bg-blue-600
+      dark:hover:bg-blue-700
+      dark:focus:ring-blue-800
+    "
+    type="button"
+  ><%= label %></button>
+  <!-- Main modal -->
+  <div
+    data-viral--modal-component-target="modal"
+    tabindex="-1"
+    aria-hidden="true"
+    class="
+      fixed
+      top-0
+      left-0
+      right-0
+      z-50
+      hidden
+      w-full
+      p-4
+      overflow-x-hidden
+      overflow-y-auto
+      md:inset-0
+      h-[calc(100%-1rem)]
+      max-h-full
+    "
+  >
+    <div class="relative w-full <%= size %> max-h-full">
+      <!-- Modal content -->
+      <div class="relative bg-white rounded-lg shadow dark:bg-gray-700">
+        <!-- Modal header -->
+        <div
+          id="Viral-Modal-Header"
```

Comment:
**nitpick (non-blocking):** Try to avoid setting element id's unless there is no other option.  The only time we have been using the `Viral-` class is if a style was created in `application.tailwind.css` which you can freely do.  What we are trying to keep to is the syntax `Viral-`**Component**`--`**element**. E.g. `Viral-Icon--colorSuccess`.

Reply:
Please let me know what you want the ids to be to make ui testing easier. In the last IRIDA project the standard was that it should be started with "t-".

Reply:
I used developer tools to inspect the modal body, then copied the xpath and selector options. This is what it returned:

xpath
/div/div/div/div[2]

selector
div > div > div > div.p-6.space-y-6.text-gray-500.dark\:text-gray-400

I liked xpath, but assert_selector uses css selectors. I changed it to div > div > div > div:nth-child(2).

Please see f549f635c1b8a63fdf655ffd8f2fe375284ce34c.

Reply:
Instead of relying on the dom structure, I think it can be simplified to just the `assert_text` line. We are already limiting the asserts to the content of the modal with the `within` statement. In general we only need to assert that the specific content is visible to the user, as they do not necessarily care that it is in a div just that it is visible.

Reply:
Yup, please see the making test less specific commit, f07000c01022cbf43440954ed68fe3bf02078196.